tlsutil/config_test.go

```diff
@@ -538,17 +533,45 @@ func TestConfigurator_CommonTLSConfigGetClientCertificate(t *testing.T) {
 	c, err := NewConfigurator(Config{}, nil)
 	require.NoError(t, err)
 
-	cert, err := c.commonTLSConfig(false).GetCertificate(nil)
+	cert, err := c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
-	require.Nil(t, cert.Certificate)
+	require.Nil(t, cert)
 
-	c.manual.cert = &tls.Certificate{}
-	cert, err = c.commonTLSConfig(false).GetCertificate(nil)
+	c1, err := loadKeyPair("../test/key/something_expired.cer", "../test/key/something_expired.key")
+	require.NoError(t, err)
+	c.manual.cert = c1
+	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
 	require.Equal(t, c.manual.cert, cert)
 
+	c2, err := loadKeyPair("../test/key/ourdomain.cer", "../test/key/ourdomain.key")
+	require.NoError(t, err)
+	c.autoEncrypt.cert = c2
 	cert, err = c.commonTLSConfig(false).GetClientCertificate(nil)
 	require.NoError(t, err)
+	require.Equal(t, c.autoEncrypt.cert, cert)
+}
+
+func TestConfigurator_CommonTLSConfigGetCertificate(t *testing.T) {
+	c, err := NewConfigurator(Config{}, nil)
+	require.NoError(t, err)
+
+	cert, err := c.commonTLSConfig(false).GetCertificate(nil)
+	require.NoError(t, err)
+	require.Nil(t, cert)
+
+	c1, err := loadKeyPair("../test/key/something_expired.cer", "../test/key/something_expired.key")
+	require.NoError(t, err)
+	c.autoEncrypt.cert = c1
+	cert, err = c.commonTLSConfig(false).GetCertificate(nil)
+	require.NoError(t, err)
+	require.Equal(t, c.autoEncrypt.cert, cert)
+
+	c2, err := loadKeyPair("../test/key/ourdomain.cer", "../test/key/ourdomain.key")
```

Comment:
nit: this test looks good now but it might not be obvious to future us exactly what the signficance f this behaviour is.

Maybe add comments that adding one cert as the auto_encrypt one gets returned as the regular _server_ cert above and then that loading a different one _as well_ as the "manual" cert means we still server the manual one not the auto-encrypt one.